test_runner/pg_clients/test_pg_clients.py

```diff
@@ -0,0 +1,51 @@
+import os
+import shutil
+import subprocess
+from pathlib import Path
+from tempfile import NamedTemporaryFile
+from urllib.parse import urlparse
+
+import pytest
+from fixtures.neon_fixtures import RemotePostgres
+
+
+@pytest.mark.remote_cluster
+@pytest.mark.parametrize(
+    "client",
+    [
+        "csharp/npgsql",
+        "java/jdbc",
+        "python/asyncpg",
+        pytest.param("python/pg8000", marks=pytest.mark.xfail(reason="Doesn't support SNI")),
+        pytest.param("swift/PostgresClientKit",
+                     marks=pytest.mark.xfail(reason="Doesn't support SNI")),
```

Comment:
Does it work with connection options `'&options=project%3D<project-name>'`, as suggested by our error message?

Reply:
As far as I can see the library doesn't support connection string or any kind of additional options [1]; but there's no validation for these parameters, so we could try to inject it into the StartupRequest

- [1] https://github.com/codewinsdotcom/PostgresClientKit/blob/v1.4.3/Sources/PostgresClientKit/ConnectionConfiguration.swift
- [2] https://github.com/codewinsdotcom/PostgresClientKit/blob/v1.4.3/Sources/PostgresClientKit/Request/StartupRequest.swift#L43-L66

Reply:
injection is sometimes the only solution, but in this case it seems especially brittle, snice it wouldn't be surprising if they added validation. Let's leave a comment with this context, to explain that they don't have SNI **and** they don't support parameters